tests/drivers/adc/adc_api/boards/stm32h573i_dk.overlay

```diff
@@ -1,9 +1,28 @@
 /*
  * Copyright (c) 2023 STMicroelectronics
+ * Copyright (c) Benjamin Björnsson <benjamin.bjornsson@gmail.com>
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+
+#include <zephyr/dt-bindings/adc/adc.h>
+
+/ {
+	zephyr,user {
+		/* adjust channel number according to pinmux in board.dts */
+		io-channels = <&adc1 1>;
+	};
+};
+
 &adc1 {
-	/delete-property/ pinctrl-0;
-	pinctrl-0 = <&adc1_inp0_pa0>; /* Arduino A2 */
```

Comment:
This is likely still required.

Reply:
I may have misunderstood this, but this pinctrl should be paired with channel 0 right?

I focused on moving the channel settings as is for clarity, but I realized that lots of boards have incorrect or missing pinctrls. I’m looking at correcting this in a future PR.